File: LeetCode/Python/#1 Two Sum.py
def twoSum(nums, target):
    """
    :type nums: List[int]
    :type target: int
    :rtype: List[int]
    """
    arr = {}
    for i in range(0,len(nums)):
        if nums[i] in arr:

            return [arr[nums[i]],i]
        else:
            arr[target - nums[i]] = i

twoSum([0,5,5,10],10)

File: LeetCode/JavaScript/#1 Two Sum.js
var twoSum = function(nums, target) {
    var len = nums.length;
   for(var i = 0;i<len;i++){
   		for(var j =i+1;j<len;j++){
   			if(nums[i]+nums[j]==target){
   				return [i,j];
   			}
   		}
   }
}

// new Way
const twoSum = function(nums, target) {
	const condition = {};
	for(let i = 0; i < nums.length; i++){
		if(condition[nums[i]] >= 0){
			return [condition[nums[i]], i];
		}
		condition[target - nums[i]] = i;
	}
}
